### ISSUE TITLE
Trying to build python-hwdata (for Spacewalk) fails with v18.11.1

### ISSUE THREAD

**Issue by rpasche** [ascherer/debbuild#108]
Trying to build the python-hwdata package within Spacewalk with v18.11.1 fails with the following error

```
root@0000000:~/debbuild/SPECS# LANG=C debbuild -ba -vv python-hwdata.spec
This is debbuild, version 18.11.1
Unknown tag '%py' at line 134 of python-hwdata.spec
root@0000000:~/debbuild/SPECS#
```

These are the lines (line numbers from vim)

```
....
130 %endif # with python3
131
132 %build
133 %if %{with python2}
134 %py2_build
135 %endif # with python2
136
137 %if %{with python3}
...
```

So the %py2_build does somehow not work (exist)

Corresponding files within the .spec file are

```
...
44 %{!?py2_build: %global py2_build CFLAGS="%{optflags}" %{__python2} setup.py build}
45 %{!?py2_install: %global py2_install %{__python2} setup.py install %{?pyinstflags} --skip-build --root %{buildroot} %{?pytargetflags}}
46 %{!?py3_build: %global py3_build CFLAGS="%{optflags}" %{__python3} setup.py build}
47 %{!?py3_install: %global py3_install %{__python3} setup.py install %{?pyinstflags} --skip-build --root %{buildroot} %{?pytargetflags}}
48
49 %{!?py3dir: %global py3dir %{_builddir}/python3-%{name}-%{version}-%{release}}
...
```

From the spacewalk-user list I heard, that this still works with version v18.8.1. I will now try that and report.

**Comment by rpasche** [comment]
I can confirm. With version (suggested by Neal within spacewalk-user list), the build works. Did not yet try other versions.

**Comment by ascherer** [comment]
Do you work with this repo: https://github.com/xsuchy/python-hwdata ?
Apart from the invocation, I can't find a definition of `%py2_build`.

`rpmbuild -E %py2_build` does not know about this macro either, so I wonder where it is defined.

**Comment by rpasche** [comment]
Yes...it is this repo. Latest version

**Comment by rpasche** [comment]
Oh ah......I think I got it. Maybe I used the .spec from the spacewalk repo. Let me check this

**Comment by rpasche** [comment]
Well...with version 18.11.1 and the original spec (from python-hwdata repo), I get nearly same error

```
root@00000:~/debbuild/SPECS# LANG=C debbuild -ba -vv python-hwdata.spec 
This is debbuild, version 18.11.1
Unknown tag '%py' at line 77 of python-hwdata.spec
root@00000:~/debbuild/SPECS#
```
And this is

```
 74 
 75 %build
 76 %if %{with python2}
 77 %py2_build
 78 %endif # with python2
 79 
 80 %if %{with python3}
 81 pushd %{py3dir}
```

And correct, the py2_build is completly missing.

But the over way round still exists. Using the "spacewalk" python-hwdata.spec file and running v18.11.1 results in mentioned error, while running v18.8.1 succeeds.

**Comment by rpasche** [comment]
I'm using this spec file https://github.com/spacewalkproject/spacewalk/blob/master/client/debian/python-hwdata/python-hwdata.spec

**Comment by ascherer** [comment]
Will `bisect` ASAP.

**Comment by rpasche** [comment]
I'm not in a hurry ;-) Found another problem. Not here, but within spacewalk Packages.gz creation (just because I created packages with your tool) :-)

**Comment by ascherer** [comment]
My best guess at this time: around line 485 grabbing `$1` as `[a-z_]+` falls short at `2` and should be replaced by `\w+`. Will try RSN.

**Comment by ascherer** [comment]
Fixed in release 18.12.0. Thanks for the report.

**Comment by ascherer** [comment]
Without `bisect`ing: the `%[a-z_]+` part of `%py2_build` matches `copyrightdata`.
